core/domain/collection_domain_test.py

```diff
@@ -824,88 +363,30 @@ def test_nodes_are_in_playble_order(self):
-        # the playable order will be equal to the order in which they are
-        # adeed.
         sorted_nodes = collection.get_nodes_in_playable_order()
-        expected_explorations_list = ['exp_id_1', 'exp_id_2', 'exp_id_0']
-        observed_explration_list = [
-            node.exploration_id for node in sorted_nodes]
-
-        self.assertEqual(expected_explorations_list, observed_explration_list)
-
-        # Add skills to collection.
-        collection.add_skill('skill0')
-        collection.add_skill('skill1')
-        collection.add_skill('skill2')
-
-        # Updating 1st node's prerequisite and acquired skill_ids.
-        collection_node0 = collection.get_node('exp_id_1')
-        collection_node0.update_prerequisite_skill_ids(['skill0'])
-        collection_node0.update_acquired_skill_ids(['skill1'])
-
-        # The 1st node will be removed from the playable order as this node
-        # is not acheivable i.e, there is no way to fullfil the prerequisite
-        # of this node.
-        sorted_nodes = collection.get_nodes_in_playable_order()
-        expected_explorations_list = ['exp_id_2', 'exp_id_0']
-        observed_explration_list = [
-            node.exploration_id for node in sorted_nodes]
-
-        self.assertEqual(expected_explorations_list, observed_explration_list)
-
-        # Updating 3rd node's prerequisite and acquired skill_ids.
-        collection_node2 = collection.get_node('exp_id_0')
-        collection_node2.update_prerequisite_skill_ids([])
-        collection_node2.update_acquired_skill_ids(['skill0'])
-
-        # Now the 1st node's prerequisite skills can be acheived through the
-        # 3rd node (exp_id_0 ---> exp_id_1).
-        sorted_nodes = collection.get_nodes_in_playable_order()
-        expected_explorations_list = ['exp_id_2', 'exp_id_0', 'exp_id_1']
-        observed_explration_list = [
-            node.exploration_id for node in sorted_nodes]
-        self.assertEqual(observed_explration_list, expected_explorations_list)
-
-        # Updating 2nd node's prerequisite and acquired skill_ids.
-        collection_node1 = collection.get_node('exp_id_2')
-        collection_node1.update_prerequisite_skill_ids(['skill1'])
-        collection_node1.update_acquired_skill_ids(['skill2'])
-
-        # Sorting nodes in linear and playable order.
-        sorted_nodes = collection.get_nodes_in_playable_order()
-
-        # Expected order of explorations.
         expected_explorations_list = ['exp_id_0', 'exp_id_1', 'exp_id_2']
-
-        # Observed order of exploration.
         observed_explration_list = [
             node.exploration_id for node in sorted_nodes]
 
-        # Validates the number of nodes present in collection after sorting.
-        self.assertEqual(len(sorted_nodes), 3)
-
-        # Checks the order of exploration.
-        self.assertEqual(observed_explration_list, expected_explorations_list)
+        self.assertEqual(expected_explorations_list, observed_explration_list)
```

Comment:
observed_explorations_list (or better, expected_exploration_ids, observed_exploration_ids). Nice to differentiate between explorations and their IDs.

Reply:
Done.